### PR TITLE
refactor(accordion): use reusable panels

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: src/accordion/accordion.spec.ts [hunk 1]
```diff
@@ -123,41 +121,53 @@ describe('ngb-accordion', () => {
        });
      }));
 
-  it('should only open one at a time', injectAsync([TestComponentBuilder], (tcb) => {
-       return tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-         const tc = fixture.componentInstance;
-         tc.closeOthers = true;
-         fixture.detectChanges();
+  xit('should only open one at a time', injectAsync([TestComponentBuilder], (tcb) => {
```
Comment (Foxandxss) [inline src/accordion/accordion.spec.ts]: no xit.

**Comment by wesleycho** [comment]
Is there a particular reason for the removal of collapse?

**Comment by Foxandxss** [comment]
I think he wants the ngbPanel to be the collapse (they do the same thing) but I think this PR is half baked or lacks proper explanation :P

**Comment by pkozlowski-opensource** [comment]
Here is some more context about the proposed refactor: if you look at the HTML structure of bs4 accordion, you will notice that tis is just a bunch of things called "panels". A panel can have a header + title, collapsible content and (optionally) a footer. We haven't had "panel" as a stand-alone directive but this PR introduces one. The reasoning here is that we should be building higher-level components from the lower-level building blocks.

Now, a panel can be collapsible and we might have it as a separate directive as well - this is one of the things I wanted to debate.

File: src/accordion/accordion.spec.ts (hunk 1 above)
Comment (pkozlowski-opensource) [inline src/accordion/accordion.spec.ts]: yeh, for whatever reason I didn't mange to make this test behave as expected.... On it right now...

**Comment by pkozlowski-opensource** [comment]
@wesleycho @Foxandxss this should be ready for review, I'm done with all the code changes and test. Hopefully my previous explanations give a bit more context but please let me know if you need more info.

**Comment by wesleycho** [comment]
I guess I'm fine with this if this is the way we want to go - I am worried about panels potentially becoming a mega component though.

**Comment by pkozlowski-opensource** [comment]
> I am worried about panels potentially becoming a mega component though.

I'm totally cool with extracting the `collapse` directive as we used to have. But I would like to be able to use panels stand-alone as well.

**Comment by Foxandxss** [comment]
Why remove isOpen from the panels?

And I think I like our collapse, we could need it for other stuff.

**Comment by pkozlowski-opensource** [comment]
> Why remove isOpen from the panels?

It was rename to `open` but otherwise still part of a panel: https://github.com/ng-bootstrap/core/pull/167/files#diff-b50f479a3cf0e488ce11076b561c5993R50

> And I think I like our collapse, we could need it for other stuff.

OK, will add it back. Out of curiosity - what is the "other stuff" you've got on your mind?

**Comment by Foxandxss** [comment]
The diffs annoy me. Yeah, I see it now.

Other stuff... Well, isn't there any other component that could be collapsed?

**Comment by pkozlowski-opensource** [comment]
@Foxandxss `NgbCollapse` is back. Anything else or is this one good to go?

**Comment by Foxandxss** [comment]
Yeah, merge it. I wanted to work with this a bit to add more tests or something (I have a few things in my TODO).

**Pull request closed without merging**